src/main/java/sirius/db/mixing/EntityDescriptor.java

```diff
@@ -1045,6 +1047,35 @@ public Property getProperty(String property) {
         return prop;
     }
 
+    /**
+     * Get the value of a mapping for an entity.
+     * <p>
+     * If the mapping contains joins, they are only resolved if (a) they are already fetched from the database, or (b)
+     * the parameter value of {@code resolveJoins} evaluates to {@code true}. In all other cases, an Exception is thrown.
+     * <p>
+     * Usually, this can be achieved using <p>
+     * {@link #getProperty(Mapping)}{@code .}{@link Property#getValue(Object) getValue(Object)}<p>
+     * However, no property can be created if the mapping is a JOIN-mapping, rendering this method useless in that case.
+     *
+     * @param mapping      the mapping; may contain joins
+     * @param entity       the entity
+     * @param resolveJoins whether to load joined columns from the database instead of throwing
+     * @return the value of the mapping
+     */
+    public Object getPropertyValue(Mapping mapping, Object entity, boolean resolveJoins) {
+        if (mapping.getParent() != null) {
+            Object parent = getPropertyValue(mapping.getParent(), entity, resolveJoins);
+            return mixing.getDescriptor(parent.getClass())
+                         .getPropertyValue(Mapping.named(mapping.getName()), parent, resolveJoins);
+        }
+        Property property = getProperty(mapping.getName());
+        if (property instanceof BaseEntityRefProperty<?, ?, ?> ref) {
+            BaseEntityRef<?, ?> entityRef = ref.getEntityRef(entity);
+            return resolveJoins ? entityRef.fetchValue() : entityRef.getValueIfPresent().orElseThrow();
```

Comment:
Throw exception with specific cause saying query not build up properly as joined mapping not resolved

Reply:
It's not by design a query that was not properly joined - the method can be used from anywhere, not just `#streamBlockwise`. However, a sensible error message does indeed make sense here.